code-collab-app/src/storage/peer_data.js

```diff
@@ -68,71 +109,150 @@ const peer_data = observable({
             };
 
             //  Request list of peers in <session_name> session
+            console.log('joining existing session...');
             this.tracker.emit('join_session', peer_info);
-
-            //  wait for a list of peers currently in the session
-            this.tracker.on('peer_list', list => {
-                this.session_peers = list;
-                this.request_doc_data();
-            });
         });
     },
 
-    //  request for current session's doc data 
-    //  from a fellow peer
+    //  request doc data from a peer
     request_doc_data() {
```

Comment:
I might be missing something here, but how does the peer know which document to request if we don't provide the document name, etc?

Reply:
When we have joined a session, all the peers only have access to one document. So we don't really need to know which document to get because there is only one in each session.

This is called after we have joined a session, and aimed towards one of the peers in the session.

Reply:
Ok so when does a peer join a session? Is this after the end user clicks on one of the "Active Sessions" on the homepage? If so, how do we guarantee the editor page will have the correct data?

Reply:
Yes, when the user clicks on one of the existing sessions, we join that session. We get a list of peers from the tracker for that session and then ask for data from the first peer. If no peers exists in the list that was returned by the tracker, then we ask the tracker for the document data using the session id.

Reply:
Oh ok I see now. ! missed the "join_session" line from the homepage so I was wondering how the peer would know a user joined but it makes sense now that I saw that line.

Reply:
Perfect :)